elastic/apm-server
# Introduce `apm-server.auth.*` config

## Motivation/summary

Introduce the new AgentAuth config structure, which holds API Key and secret token auth. Later we will add "anonymous" auth here too, and deprecate/replace some RUM config (rate limiting and allowed service names).

We also introduce a new YAML naming scheme for the config, `apm-server.auth.*`. The old config is deprecated and copied across to the new config fields.

## Checklist


## How to test these changes

1. Set `apm-server.secret_token` and `apm-server.api_key.*`, make sure they are honoured (e.g. query "GET /" with/out auth)
2. Set `apm-server.auth.secret_token` and `apm-server.auth.api_key.*`, same again.
3. Set both new and old with different values: check that a warning is logged that the old config is ignored, and check that it is possible to auth using the new but not old

## Related issues

https://github.com/elastic/apm-server/issues/5347
